### PR TITLE
`Vector4`/`Vector4i`: Add missing methods and tests

### PR DESCRIPTION
**Pull request by Geometror**
- Implement the following methods for Vektor4:
	 `get_axis, set_axis, set_all, cubic_interpolate, posmod, posmodv, snap, snapped, distance_to, direction_to`
- Add basic tests for Vektor4 and Vektor4i
	- Vektor4: Fix operator-()

### CODE REVIEW

File: core/math/vector4.cpp [hunk 1]
```diff
@@ -53,6 +87,16 @@ bool Vector4::is_normalized() const {
 	return Math::is_equal_approx(length_squared(), 1, (real_t)UNIT_EPSILON); // Use less epsilon.
 }
 
+real_t Vector4::distance_to(const Vector4 &p_to) const {
```
Comment (reduz) [inline core/math/vector4.cpp:90]: I feels these are entirely useless, but if it makes you feel better I don't object to merging them.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
When fixing the CI issues, you can also fix the spelling of Vector4i in the commit for easier future grepping.

**Comment by Geometror** [comment]
Fixed :)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**